presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```diff
@@ -106,153 +131,171 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         HttpServletRequest request = (HttpServletRequest) servletRequest;
         HttpServletResponse response = (HttpServletResponse) servletResponse;
 
-        String header = request.getHeader(HttpHeaders.AUTHORIZATION);
-        if (header == null) {
-            // request for user/password for LDAP authentication
-            sendChallenge(response, SC_UNAUTHORIZED, null);
-            return;
-        }
-
-        List<String> parts = Splitter.on(WHITESPACE).splitToList(header);
-        if (parts.size() != 2) {
-            sendChallenge(response, SC_BAD_REQUEST, "Invalid value for authorization header");
-            return;
-        }
-
-        String authenticationType = parts.get(0);
-        if (!authenticationType.equalsIgnoreCase(AUTHENTICATION_TYPE)) {
-            sendChallenge(response, SC_BAD_REQUEST, format("Incorrect authentication type (expected %s): %s", AUTHENTICATION_TYPE, authenticationType));
-            return;
-        }
-
-        String[] credentials = new String(Base64.getDecoder().decode(parts.get(1))).split(":", 2);
-        if (credentials.length != 2) {
-            sendChallenge(response, SC_BAD_REQUEST, "Username/password missing in the request header");
-            return;
-        }
-
-        DirContext context = null;
-        String user = credentials[0];
-
         try {
-            String password = credentials[1];
-            if (user.isEmpty() || password.isEmpty()) {
-                throw new AuthenticationException("Username or password is empty");
-            }
-
-            Hashtable<String, String> environment = getBasicEnvironment();
-            String principal = genericLdapBinder.getUserBindSearchPattern().replaceAll("\\$\\{USER\\}", user);
```

Comment:
It is there. I extracted it to the separate method  `private String createPrincipal(String user)`